src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -146,11 +147,19 @@ private DashboardCommands parseDashboardCommandFromInput(String input) {
      * @param input full user input string
      * @return module code string
      */
-    private String getModuleCode(String input) {
-        // TODO  - error handling
+    private String getModuleCode(String input) throws CommandException {
         String[] words = input.split(" ");
+        if (words.length < 2) {
+            throw new CommandException("Module not specified.");
+        }
+
+        String moduleCode = words[1].toUpperCase();
```

Comment:
Again magic number to a constant.

Reply:
Will remember to update in next PR, thanks!